swarm/src/behaviour.rs

```diff
@@ -127,10 +129,10 @@ pub trait NetworkBehaviour: Send + 'static {
     /// has previously been called with this `PeerId`.
     fn inject_event(
         &mut self,
-        peer_id: PeerId,
-        connection: ConnectionId,
-        event: <<Self::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::OutEvent
-    );
+        _: PeerId,
+        _: ConnectionId,
+        _: <<Self::ProtocolsHandler as IntoProtocolsHandler>::Handler as ProtocolsHandler>::OutEvent
+    ) { }
```

Comment:
A `NetworkBehaviour` that does not receive any events from its `ProtocolsHandler` should use `void::Void` on their `ProtocolsHandler::OutEvent`, thus checked at compile time, no?

I would prefer the above over a default implementation. What do you think? If you agree, could you add a doc comment to `inject_event`?

Reply:
We can do that yeah.

I approached this more from a prototyping perspective: If I want to define a new `NetworkBehaviour`, what is the least amount of stuff I am faced with?

I was also tempted to default `poll` actually. An `AddressBook` style `NetworkBehaviour` doesn't need `poll` for example :)

I am not sure `inject_event` with `match event { }` is actually cleaner. The only thing it protects you from is forgetting to deal with your events if you change the `ProtocolsHandler`.

For this to work, it implies that people actually override it with `match event { }` in the first place and not just leave it blank. At least rust-analyzer autocompletes trade-impls with `todo!()` though and the path of least resistance is to either leave those in or delete them 🤷‍♂️

In any case, defaulting the others is already an improvement so I'll do that.